File: apps/mobile/targets/positions-widget/Sources/DesignSystem/Sizing.swift
import CoreGraphics

/// Chrome dimensions that are neither spacing (gaps between elements)
/// nor typography. Anything that constrains a `frame`, sets a stroke
/// thickness, or sizes an icon belongs here.
enum Sizing {
  enum RangeBar {
    static let track: CGFloat = 8
    static let thumb: CGFloat = 16
    static let thumbStroke: CGFloat = 2.5
    static let thumbShadowRadius: CGFloat = 6
  }

  enum ReverseButton {
    static let small: CGFloat = 18
    static let medium: CGFloat = 20
    /// SF Symbol point size for the small widget reverse glyph.
    static let smallIconWeight: CGFloat = 9
    /// SF Symbol point size for the medium widget reverse glyph.
    static let mediumIconWeight: CGFloat = 10
  }

  enum Tag {
    static let dotSmall: CGFloat = 5
    static let dotLarge: CGFloat = 6
  }

  enum Divider {
    static let thin: CGFloat = 1
  }

  enum Icon {
    /// SF Symbol point size for the empty-state placeholder icon.
    static let emptyState: CGFloat = 22
  }
}

File: apps/mobile/modules/widget-bridge/android/src/main/java/expo/modules/widgetbridge/widget/theme/Sizing.kt
package expo.modules.widgetbridge.widget.theme

import androidx.compose.ui.unit.dp

object Sizing {
  object RangeBar {
    val track = 14.dp
    val thumb = 24.dp
    // Bitmap-space pixel stroke around the thumb circle (RangeBarView renders
    // into a 600x48 px bitmap; this is in those pixel units, not dp).
    const val thumbStrokePx = 7f
    val thumbShadowRadius = 6.dp
  }

  object ReverseButton {
    val small = 18.dp
    val medium = 20.dp
    val smallIcon = 9.dp
    val mediumIcon = 10.dp
  }

  object Tag {
    val dotSmall = 5.dp
    val dotLarge = 6.dp
  }

  object Divider {
    val thin = 1.dp
  }

  object Icon {
    val emptyState = 22.dp
  }

  object WidgetMetrics {
    val contentMargin = 16.dp
  }
}
